network/src/peers/peer_book.rs

```diff
@@ -265,31 +269,22 @@ impl PeerBook {
     }
 
     ///
-    /// Returns a reference to the peer info of the given address, if it exists.
+    /// Returns the peer info of the given address, if it exists.
     ///
-    pub fn get_peer(&self, address: SocketAddr) -> Result<PeerInfo, NetworkError> {
+    pub fn get_peer(&self, address: SocketAddr, only_if_connected: bool) -> Option<PeerInfo> {
         // Check if the address is a connected peer.
         if self.is_connected(address) {
             // Fetch the peer info of the connected peer.
-            return self
-                .connected_peers()
-                .get(&address)
-                .cloned()
-                .ok_or(NetworkError::PeerBookMissingPeer);
+            return self.connected_peers().get(&address).cloned();
         }
 
         // Check if the address is a known disconnected peer.
-        if self.is_disconnected(address) {
+        if !only_if_connected && self.is_disconnected(address) {
```

Comment:
Isn't this check already covered by `self.is_connected`?

Reply:
in this particular case yes, but I'd like to keep that extra param for the future, in case either option is desired